client/signup/steps/woocommerce-install/step-store-address/index.tsx

```diff
@@ -162,3 +212,56 @@ export default function StepStoreAddress( props: WooCommerceInstallProps ): Reac
 		/>
 	);
 }
+
+function ControlError( props: { error: string } ): ReactElement | null {
+	const { error } = props;
+	if ( error ) {
+		return <FormInputValidation isError={ true } isValid={ false } text={ error } />;
+	}
+	return null;
+}
+
+function useAddressFormValidation( siteId: number ) {
```

Comment:
I don't think using a hook for this makes sense, I was trying something out and it made its way into production. 

Ideally we clean this up next sprint. I think all this state should get moved up out of the site settings hook and the error validation gets coupled in with the values they refer to.

The site settings hook might still make sense but I think the form still needs to retain control over state/value changes. The hook should only be involved on submit. Or maybe it makes sense to move this validation code into the site settings hook somehow. :thinking: